packages/flutter/lib/src/widgets/scroll_simulation.dart

```diff
@@ -165,67 +169,146 @@ class ClampingScrollSimulation extends Simulation {
-    return _distance * _flingVelocityPenetration(t) * velocity.sign / _duration;
+    if (time == 0) {
+      return velocity;
+    }
+    final _NBSample sample = _NBSample(time, _duration);
+    return sample.velocityCoef * _distance / _duration * velocity.sign * 1000.0;
   }
 
   @override
   bool isDone(double time) {
-    return time >= _duration;
+    return time * 1000.0 >= _duration;
+  }
+}
+
+class _NBSample {
+  _NBSample(double time, int duration) {
+    _initSplinePosition();
+
+    // See computeScrollOffset().
+    final double t = time * 1000.0 / duration;
+    final int index = (_nbSamples * t).round();
+    _distanceCoef = 1.0;
+    _velocityCoef = 0.0;
+    if (index < _nbSamples) {
+      final double tInf = index / _nbSamples;
+      final double tSup = (index + 1) / _nbSamples;
+      final double dInf = _splinePosition[index];
+      final double dSup = _splinePosition[index + 1];
+      _velocityCoef = (dSup - dInf) / (tSup - tInf);
+      _distanceCoef = dInf + (t - tInf) * _velocityCoef;
+    }
+  }
+
+  late double _velocityCoef;
+  double get velocityCoef => _velocityCoef;
+
+  late double _distanceCoef;
+  double get distanceCoef => _distanceCoef;
+
+  static const int _nbSamples = 100;
+  static final List<double> _splinePosition =
+      List<double>.filled(_nbSamples + 1, 0.0);
+  static final List<double> _splineTime =
+      List<double>.filled(_nbSamples + 1, 0.0);
+  static const double _startTension = 0.5;
+  static const double _endTension = 1.0;
+  static bool _isInitialized = false;
+
+  // See static iniitalization in Scroller.java.
+  static void _initSplinePosition() {
+    if (_isInitialized) {
+      return;
+    }
+    const double p1 = _startTension * _inflexion;
+    const double p2 = 1.0 - _endTension * (1.0 - _inflexion);
+    double xMin = 0.0;
+    double yMin = 0.0;
+    for (int i = 0; i < _nbSamples; i++) {
+      final double alpha = i / _nbSamples;
+      double xMax = 1.0;
+      double x, tx, coef;
+      while (true) {
+        x = xMin + (xMax - xMin) / 2.0;
+        coef = 3.0 * x * (1.0 - x);
+        tx = coef * ((1.0 - x) * p1 + x * p2) + x * x * x;
+        if ((tx - alpha).abs() < 1e-5) {
+          break;
+        }
+        if (tx > alpha) {
+          xMax = x;
+        } else {
+          xMin = x;
+        }
+      }
+      _splinePosition[i] = coef * ((1.0 - x) * _startTension + x) + x * x * x;
+      double yMax = 1.0;
+      double y, dy;
+      while (true) {
+        y = yMin + (yMax - yMin) / 2.0;
+        coef = 3.0 * y * (1.0 - y);
+        dy = coef * ((1.0 - y) * _startTension + y) + y * y * y;
+        if ((dy - alpha).abs() < 1e-5) {
+          break;
+        }
+        if (dy > alpha) {
+          yMax = y;
+        } else {
+          yMin = y;
+        }
+      }
+      _splineTime[i] = coef * ((1.0 - y) * p1 + y * p2) + y * y * y;
+    }
+    _splinePosition[_nbSamples] = _splineTime[_nbSamples] = 1.0;
+    _isInitialized = true;
```

Comment:
can we do this statically and just fill in the data? (rather than doing it at runtime on first use)

Reply:
Do you mean statically through code, like in the JAVA implementation? If so, no I don't think Dart offers anything similar hence the somewhat awkward implementation.

If you actually meant writing an external program to create the static data we could do that. It would obviously not be as easily configured liked the Android implementation.

Reply:
There is no "static initialization" time in Dart. All code that runs should do so as a result of invoking `main()`.

The options here are either to compute it on first use, or to add a hook somewhere in the framework to cause it to be computed early.

Reply:
I meant offline (as in have a program in the `dev/` directory that runs this code and generates the constants and you can just paste them into this file). Looks like this code isn't configurable by developers, so that wouldn't reduce the overall configurability for our developers, but would be more efficient for users.

Reply:
Done.